lib/spack/docs/basic_usage.rst

```diff
@@ -1234,6 +1242,88 @@ add a version specifier to the spec:
 Notice that the package versions that provide insufficient MPI
 versions are now filtered out.
 
+
+-----------------------------
+Deprecating insecure packages
+-----------------------------
+
+``spack deprecate`` allows for the removal of insecure packages with
+minimal impact to their dependents.
+
+.. warning::
+
+  The ``spack deprecate`` command is designed for use only in
+  extraordinary circumstances. This is a VERY big hammer to be used
+  with care.
```

Comment:
It feels a bit strange to have a warning that says:
```
The ``spack deprecate`` command is designed for use only in extraordinary circumstances
```
under the "Basic usage" section of the docs. Wondering if we should move this subsection elsewhere (here or in another PR).

Reply:
This is part of why I moved it to the end.  I think we need a larger rework but for now I just moved some sections around.